src/thunderbots/software/util/parameter/generate_parameters.py

```diff
@@ -0,0 +1,291 @@
+            fpe.write(constants.CFG_HEADER)
+
+            # generate cfg
+            __cfg_gen(param_info[key], fpe)
+
+            # write footer
+            fpe.write(constants.CFG_FOOTER.format(key))
+
+            # make file read/write/executable
+            os.chmod(output_path+key+constants.EXT, 0o0754)
+            print '=== generated {} ==='.format(key)
+
+
+def __cfg_gen(param_info, file_pointer, group_name=None):
+    """Takes the information about parameters and namespaces,
+    and recursively generates the configuration
+
+    :param param_info: Contains namespace and parameter information
+        organized by namespaces until the parameter
+    :param file_pointer: The file to store the values
+    :param group_name: The name of the current group to add to
+    :type param_info: dict
+    :type file_pointer: file
+    :type group_name: str
+
+    """
+    # iterate through keys
+    for key in param_info.keys():
+
+        # if type key is found, create a parameter
+        if "type" in param_info[key]:
+            try:
+                parameter = param_info[key]
+
+                # write parameter to file
+                file_pointer.write(
+                    constants.CFG_PARAMETER.format(
+                        namespace=group_name if group_name is not None else "gen",
+                        name=key,
+                        type=constants.CFG_TYPE_MAP[parameter["type"]],
+                        description=parameter["description"],
+                        default=parameter["default"],
+                        quote="\"" if parameter["type"] in constants.QUOTE_TYPES else "",
+                        min_val=parameter["min"] if parameter["type"] in constants.RANGE_TYPES else None,
+                        max_val=parameter["max"] if parameter["type"] in constants.RANGE_TYPES else None
+                    ))
+
+            except KeyError as kerr:
+                print constants.AUTOGEN_FAILURE_MSG
+                sys.exit('Required key missing from config: {}, please check parameter {}'.format(
+                    kerr, key))
+
+        # if there is no current group, a new namespace must be created
+        elif group_name is None:
+            file_pointer.write(constants.CFG_NEW_NAMESPACE.format(
+                namespace=key
+            ))
+            __cfg_gen(param_info[key], file_pointer, group_name=key)
+
+        # if there already is a group, add to that namespace
+        else:
+            file_pointer.write(constants.CFG_SUB_NAMESPACE.format(
+                namespace=group_name,
+                sub_namespace=key
+            ))
+            __cfg_gen(param_info[key], file_pointer, group_name=key)
+
+#######################################################################
+#                            CPP Generator                            #
+#######################################################################
+
+
+def generate_header_cpp(param_info, output_path):
```

Comment:
Took me a second to realize that this meant generate the header _and_ the cpp, rather then the cpp header, perhaps this might be a bit more clear?:
``` suggestion
def generate_header_and_cpp(param_info, output_path):
```

Reply:
Fixed